core/src/main/scala/com/pingcap/tispark/PartitionExpr.scala

```diff
@@ -0,0 +1,205 @@
+package com.pingcap.tispark
+
+import java.util
+
+import com.google.common.collect.Range
+import com.pingcap.tikv.meta.{TiPartitionDef, TiPartitionInfo, TiTableInfo}
+import org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute
+import org.apache.spark.sql.catalyst.expressions.{And, AttributeReference, BinaryComparison, EqualTo, Expression, GreaterThan, GreaterThanOrEqual, IsNotNull, IsNull, LessThan, LessThanOrEqual, Literal, Or, UnaryExpression}
+
+case class AsOrdered[T](val value: T)(implicit ord: Ordering[T]) extends Ordered[AsOrdered[T]] {
+  override def compare(that: AsOrdered[T]) = ord.compare(value, that.value)
+  def open[T: Ordering](from: T, to: T) = {
+    val ord = implicitly[Ordering[T]]
+    Range.open(AsOrdered(from), AsOrdered(to))
+  }
+}
+
+case class PartitionExpr(partitionExprs: List[Expression],
+                         locateExprs: List[Expression],
+                         partitionExpr: Expression,
+                         col: String) {
+  val ranges = partitionExprs
+  val partExpr = partitionExpr
+  val upperBound = locateExprs
+  val columnName = col
+
+  private def buildExprRange(e: Expression): Range[AsOrdered[Long]] = {
```

Comment:
Why this cannot be in with ScanAnalyzer class? This should be with similar logic.